Content.Server/NPC/HTN/PrimitiveTasks/Operators/Interactions/UnPullOperator.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Content.Shared.Pulling;
+using Content.Shared.Pulling.Components;
+
+namespace Content.Server.NPC.HTN.PrimitiveTasks.Operators.Interactions;
+
+public sealed partial class UnPullOperator : HTNOperator
+{
+    [Dependency] private readonly IEntityManager _entManager = default!;
+    private SharedPullingSystem _pulling = default!;
+
+    public override void Initialize(IEntitySystemManager sysManager)
+    {
+        base.Initialize(sysManager);
+
+        _pulling = sysManager.GetEntitySystem<SharedPullingSystem>();
+    }
+
+    public override async Task<(bool Valid, Dictionary<string, object>? Effects)> Plan(NPCBlackboard blackboard,
+        CancellationToken token)
+    {
+        var owner = blackboard.GetValue<EntityUid>(NPCBlackboard.Owner);
+
+        if (!_pulling.IsPulled(owner) ||
+            !_entManager.TryGetComponent<SharedPullableComponent>(owner, out var pullable))
+            return (false, null);
+
+        return (_pulling.TryStopPull(pullable), null);
+    }
+
```

Comment:
One of the downsides of calling this directly is it bypasses all sort of interaction checks such as CanInteract / CanAccess which still need to be checked unless you use interactionsystem instead.

Reply:
So, which method can i use to replace this?

Reply:
You mean i need to use InteractionSystem to check for CanAccess and CanInteract?

Reply:
You should just be using either a verb or the action itself rather than calling specific systems directly as it's more re-useable and less prone to bugs.